### PR TITLE
Build VSIX project using RC3 VSSDK tooling

### PR DESCRIPTION
**Pull request by Faizan2304**

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -479,6 +449,71 @@ function PrintAndExit-OnError([System.String] $output){
...
+    $vsixVersion = "15.0.3" # Hardcode since we want to keep 15.0.0 for other assemblies.
+
+    # VersionSuffix in microbuild comes in the form preview-20170111-01(preview-yyyymmdd-buildNoOfThatDay)
+    # So Version of the vsix will be 15.0.3.2017011101
+    $vsixVersionSuffix = $VersionSuffix.Split("-");
+    if($vsixVersionSuffix.Length -ige 2) {
+        $vsixVersion = "$vsixVersion.$($vsixVersionSuffix[1])$($vsixVersionSuffix[2])"
+    }
+
+    $menifestContentWithVersion = Get-Content "$TPB_VSIX_DIR\source.extension.vsixmanifest" -raw | % {$_.ToString().Replace("`$version`$", "$vsixVersion") }
```
Comment (sbaid) [inline scripts/build.ps1]: typo menifest

File: NuGet.config [hunk 2]
```diff
@@ -6,6 +6,7 @@
     <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
     <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
     <add key="vstest" value="https://dotnet.myget.org/F/vstest/auth/967ad631-d24c-4c0e-bece-922de3064956/api/v3/index.json" />
+	<add key="dotnet.myget.org roslyn-tools" value="https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json" />
```
Comment (codito) [inline NuGet.config]: nit: indent

File: scripts/build.ps1 [hunk 3]
```diff
@@ -479,6 +449,71 @@ function PrintAndExit-OnError([System.String] $output){
     }
 }
 
+function Locate-MSBuildPath() {
```
Comment (codito) [inline scripts/build.ps1]: Please follow the coding convention. `{` on a new line. No `()` after a function.

File: NuGet.config (hunk 2 above)
Comment (codito) [inline NuGet.config]: Just `roslyn-tools` is good.

File: scripts/build.ps1 (hunk 1 above)
Comment (Faizan2304) [inline scripts/build.ps1]: Fixed in next commit

File: NuGet.config (hunk 2 above)
Comment (Faizan2304) [inline NuGet.config]: Fixed in next commit

File: src/VSIXProject/TestPlatform.csproj [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    <Compile Include="Properties\AssemblyInfo.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <!-- Localized resources -->
```
Comment (sbaid) [inline src/VSIXProject/TestPlatform.csproj]: test - 
cd %programfiles(x86)%\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\Extensions\qgyll0uu.4m0
dir /s /b > oldfilelist.txt
cd %programfiles(x86)%\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\Extensions\testplatform
dir /s /b > newfilelist.txt
see the diff of oldfilelist.txt newfilelist.txt

File: src/VSIXProject/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Reflection;
```
Comment (codito) [inline src/VSIXProject/Properties/AssemblyInfo.cs]: Is it required to have an AssemblyInfo?

File: scripts/build.ps1 [hunk 6]
```diff
@@ -479,6 +449,71 @@ function PrintAndExit-OnError([System.String] $output){
...
+   }
+
+   Write-Verbose "VSInstallPath is : $vsInstallPath"
+
+   return Resolve-Path -path $vsInstallPath
+}
+
+function Locate-LocateVsApi {
+
+  $locateVsApi = Join-Path -path $env:TP_PACKAGES_DIR -ChildPath "RoslynTools.Microsoft.LocateVS\$env:LOCATE_VS_API_VERSION\tools\LocateVS.dll"
```
Comment (codito) [inline scripts/build.ps1]: Add a verbose log for located path.

Comment (Faizan2304) [inline scripts/build.ps1]: ok

File: scripts/build.ps1 (hunk 3 above)
Comment (Faizan2304) [inline scripts/build.ps1]: Fixed in next commit

File: scripts/build.ps1 [hunk 7]
```diff
@@ -479,6 +449,71 @@ function PrintAndExit-OnError([System.String] $output){
...
+  $msbuildPath = Join-Path -path $vsInstallPath -childPath "MSBuild\$env:MSBUILD_VERSION\Bin"
+
+  return Resolve-Path -path $msbuildPath
+}
+
+function Locate-VsInstallPath() {
+
+   $locateVsApi = Locate-LocateVsApi
+
+   $requiredPackageIds = @()
```
Comment (codito) [inline scripts/build.ps1]: If we know the list of elements, there is no need to append to the collection.

$`requiredPackageIds = @('a', 'b')`

File: scripts/build.ps1 [hunk 8]
```diff
@@ -299,56 +301,24 @@ function Create-VsixPackage
...
-    # Zip the folder
-    # TODO remove vsix creator
-    $dotnetExe = Get-DotNetPath
-    & $dotnetExe restore src\Microsoft.TestPlatform.VSIXCreator\Microsoft.TestPlatform.VSIXCreator.csproj
-    & $dotnetExe build src\Microsoft.TestPlatform.VSIXCreator\Microsoft.TestPlatform.VSIXCreator.csproj
-    & src\Microsoft.TestPlatform.VSIXCreator\bin\$TPB_Configuration\net46\Microsoft.TestPlatform.VSIXCreator.exe $packageDir $env:TP_OUT_DIR\$TPB_Configuration
-
+	
+	# Build vsix project
+    Write-Verbose "Locating MSBuild install path..."
```
Comment (codito) [inline scripts/build.ps1]: nit: indent

File: scripts/build.ps1 [hunk 9]
```diff
@@ -299,56 +301,24 @@ function Create-VsixPackage
...
+    $msbuildPath = Locate-MSBuildPath
+	
+	# Build vsix project to get TestPlatform.vsix
+	
+	Write-Log ".. .. Create-VsixPackage: Building project $TPB_VSIX_DIR\TestPlatform.csproj"
```
Comment (codito) [inline scripts/build.ps1]: nit: second level log. Use `Write-Log ".. xyz"`.

File: scripts/build.ps1 [hunk 10]
```diff
@@ -299,56 +301,24 @@ function Create-VsixPackage
...
+	& $msbuildPath\msbuild.exe "$TPB_VSIX_DIR\TestPlatform.csproj" -p:Configuration=$Configuration
```
Comment (codito) [inline scripts/build.ps1]: Write-Verbose for the command line executed.

File: src/VSIXProject/TestPlatform.csproj [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+    <Content Include="License.rtf">
+      <CopyToOutputDirectory>Always</CopyToOutputDirectory>
+      <IncludeInVSIX>true</IncludeInVSIX>
+    </Content>
+    <None Include="source.extension.vsixmanifest">
+      <SubType>Designer</SubType>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="index.html" />
```
Comment (codito) [inline src/VSIXProject/TestPlatform.csproj]: Are these required?

File: src/VSIXProject/index.html [hunk 12]
```diff
@@ -0,0 +1,65 @@
+﻿<!DOCTYPE html>
```
Comment (codito) [inline src/VSIXProject/index.html]: Remove

File: src/VSIXProject/TestPlatform.csproj [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+    <Content Include="stylesheet.css" />
```
Comment (codito) [inline src/VSIXProject/TestPlatform.csproj]: I think we don't need a separate 3rd party notices in vsix? /cc @sbaid

File: src/VSIXProject/stylesheet.css [hunk 14]
```diff
@@ -0,0 +1,129 @@
+body {
```
Comment (codito) [inline src/VSIXProject/stylesheet.css]: Remove

File: src/package/external/external.csproj [hunk 15]
```diff
@@ -41,6 +41,10 @@
       <Version>14.0.12-pre</Version>
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
+	<PackageReference Include="RoslynTools.Microsoft.LocateVS">
```
Comment (codito) [inline src/package/external/external.csproj]: nit: indent

**Review by codito — APPROVED** [review]
Approved with suggestions.

File: src/VSIXProject/TestPlatform.csproj (hunk 11 above)
Comment (Faizan2304) [inline src/VSIXProject/TestPlatform.csproj]: Its coming by default when you create a vsix project.

**Comment by codito** [comment]
Looks like Jenkins machines don't have the required configuration. We should skip VSIX creation in such case instead of failing the build.

File: src/VSIXProject/TestPlatform.csproj (hunk 13 above)
Comment (sbaid) [inline src/VSIXProject/TestPlatform.csproj]: I think it is needed.

File: src/VSIXProject/index.html (hunk 12 above)
Comment (Faizan2304) [inline src/VSIXProject/index.html]: Removed in next commit

File: src/package/external/external.csproj (hunk 15 above)
Comment (Faizan2304) [inline src/package/external/external.csproj]: Fixed in next commit

File: src/VSIXProject/stylesheet.css (hunk 14 above)
Comment (Faizan2304) [inline src/VSIXProject/stylesheet.css]: Should we delete it?
It is auto generated file.

**Comment by Faizan2304** [comment]
> Looks like Jenkins machines don't have the required configuration. We should skip VSIX creation in such case instead of failing the build.
> 

Yes I am working on this.

File: scripts/build.ps1 (hunk 9 above)
Comment (Faizan2304) [inline scripts/build.ps1]: Fixed

File: scripts/build.ps1 (hunk 10 above)
Comment (Faizan2304) [inline scripts/build.ps1]: Fixed

File: scripts/build.ps1 (hunk 8 above)
Comment (Faizan2304) [inline scripts/build.ps1]: Fixed

File: src/VSIXProject/Properties/AssemblyInfo.cs (hunk 5 above)
Comment (Faizan2304) [inline src/VSIXProject/Properties/AssemblyInfo.cs]: Not necessary. We can remove it.

File: src/VSIXProject/TestPlatform.csproj (hunk 4 above)
Comment (Faizan2304) [inline src/VSIXProject/TestPlatform.csproj]: Done!!! 
Everything is same except one file CertificateInformation.dat.
I don't have info about who place it there..

File: src/VSIXProject/index.html [hunk 16]
```diff
@@ -0,0 +1,64 @@
+﻿<html>
+<head>
+    <meta charset='utf-8'>
+
+    <link rel="stylesheet" type="text/css" href="stylesheet.css" media="screen">
```
Comment (AbhitejJohn) [inline src/VSIXProject/index.html]: I don't think we need this and the css file.

File: src/VSIXProject/TestPlatform.csproj [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+    <GeneratePkgDefFile>false</GeneratePkgDefFile>
+    <IncludeAssemblyInVSIXContainer>false</IncludeAssemblyInVSIXContainer>
+    <IncludeDebugSymbolsInVSIXContainer>false</IncludeDebugSymbolsInVSIXContainer>
+    <IncludeDebugSymbolsInLocalVSIXDeployment>false</IncludeDebugSymbolsInLocalVSIXDeployment>
+    <CopyBuildOutputToOutputDirectory>false</CopyBuildOutputToOutputDirectory>
+    <CopyOutputSymbolsToOutputDirectory>false</CopyOutputSymbolsToOutputDirectory>
+    <IsProductComponent>true</IsProductComponent>
+    <TestPlatformRoot>..\..\</TestPlatformRoot>
+    <VsixInputFileLocation>$(TestPlatformRoot)\artifacts\$(Configuration)\net46\win7-x64</VsixInputFileLocation>
+    <OutputPath>$(TestPlatformRoot)\artifacts\$(Configuration)\VSIX</OutputPath>
```
Comment (AbhitejJohn) [inline src/VSIXProject/TestPlatform.csproj]: Might be interesting: Adding "<DeployExtension>False</DeployExtension>" netted us a 2 minute benefit in PR builds. You might want to add it here too. Here is a useful blog of what it means: http://blog.paranoidcoding.com/2014/06/07/deploying-vsix-to-appveyor.html

File: src/VSIXProject/index.html (hunk 16 above)
Comment (Faizan2304) [inline src/VSIXProject/index.html]: Fixed

File: src/VSIXProject/TestPlatform.csproj (hunk 17 above)
Comment (Faizan2304) [inline src/VSIXProject/TestPlatform.csproj]: Very helpful :)

**Pull request merged**